File: trustdevice/src/main/java/cn/tongdun/mobrisk/TDRisk.java
package cn.tongdun.mobrisk;

import android.content.Context;

import org.json.JSONObject;

import cn.tongdun.mobrisk.core.FMCore;

/**
 * @description: fingerprint entry class
 * @author: wuzuchang
 * @date: 2022/12/5
 */
public class TDRisk {

    public static void init(final Context context) {
        initWithOptions(context, null);
    }

    public static void initWithOptions(final Context context, Builder builder) {
        if (context == null) {
            return;
        }
        FMCore.getInstance().init(context, builder == null ? null : builder.build());
    }

    public static JSONObject getBlackbox() {

        return FMCore.getInstance().getDeviceInfo();
    }

    public static class Builder {

        private String partnerCode;
        private TDRiskCallback callback;

        public Builder partner(String partner) {
            partnerCode = partner;
            return this;
        }

        public Builder callback(TDRiskCallback TDRiskCallback) {
            callback = TDRiskCallback;
            return this;
        }

        public TDRiskOption build() {
            TDRiskOption options = new TDRiskOption();
            options.setPartnerCode(partnerCode);
            options.setCallback(callback);
            return options;
        }
    }
}

File: trustdevice/src/main/java/cn/tongdun/mobrisk/TDRisk.kt
package cn.tongdun.mobrisk

import android.content.Context
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory.Companion.instance
import cn.tongdun.mobrisk.core.FMCore
import org.json.JSONObject


/**
 * @description: fingerprint entry class
 * @author: wuzuchang
 * @date: 2023/5/12
 */
object TDRisk {

    @JvmStatic
    fun init(context: Context) {
        val builder = Builder()
        initWithOptions(context, builder)
    }

    @JvmStatic
    fun initWithOptions(context: Context, builder: Builder) {
        FMCore.instance.init(context,builder.build())
    }

    @JvmStatic
    fun getBlackbox(): JSONObject {
        return FMCore.instance.getDeviceInfo()
    }

    class Builder {
        private var partnerCode: String? = null
        private var callback: TDRiskCallback? = null
        fun partner(partner: String): Builder {
            partnerCode = partner
            return this
        }

        fun callback(TDRiskCallback: TDRiskCallback): Builder {
            callback = TDRiskCallback
            return this
        }

        fun build(): TDRiskOption {
            val options = TDRiskOption()
            options.partnerCode = partnerCode
            options.callback = callback
            return options
        }
    }

}